subsys/logging/log_backend_rpc_history_ram.c

```diff
@@ -14,6 +14,40 @@
 static uint32_t __aligned(Z_LOG_MSG_ALIGNMENT) log_history_raw[HISTORY_WLEN];
 static struct mpsc_pbuf_buffer log_history_pbuf;
 
+static bool copy_to_pbuffer(const union log_msg_generic *msg)
+{
+	size_t wlen;
+	union mpsc_pbuf_generic *dst;
+	uint8_t *dst_data;
+	uint8_t *src_data;
+	size_t hdr_wlen;
+
+	wlen = log_msg_generic_get_wlen((union mpsc_pbuf_generic *)msg);
+	dst = mpsc_pbuf_alloc(&log_history_pbuf, wlen, K_NO_WAIT);
+	if (!dst) {
+		/* No space to store the log */
+		return false;
+	}
+
+	/* First word contains internal mpsc packet flags and when copying
+	 * those flags must be omitted.
+	 */
+	dst_data = (uint8_t *)dst + sizeof(struct mpsc_pbuf_hdr);
+	src_data = (uint8_t *)msg + sizeof(struct mpsc_pbuf_hdr);
+	hdr_wlen = DIV_ROUND_UP(sizeof(struct mpsc_pbuf_hdr), sizeof(uint32_t));
+
+	if (wlen <= hdr_wlen) {
+		return false;
+	}
+
+	dst->hdr.data = msg->buf.hdr.data;
+	memcpy(dst_data, src_data, (wlen - hdr_wlen) * sizeof(uint32_t));
```

Comment:
If the rounding up was actually needed, this code would be incorrect as it would copy too few bytes. Suggest a bit more assumption-less code:
```suggestion
	if (wlen * sizeof(uint32_t) <= sizeof(struct mpsc_pbuf_hdr)) {
		return false;
	}

	dst->hdr.data = msg->buf.hdr.data;
	memcpy(dst_data, src_data, wlen * sizeof(uint32_t) - sizeof(struct mpsc_pbuf_hdr));
```